### PR TITLE
Add support of best behavior

### PR DESCRIPTION
**Pull request by j-mracek**
Requires: https://github.com/rpm-software-management/libdnf/pull/826

### CODE REVIEW

File: dnf/dnf-main.c [hunk 1]
```diff
@@ -82,8 +84,10 @@ process_global_option (const gchar  *option_name,
 
 static const GOptionEntry global_opts[] = {
   { "assumeyes", 'y', G_OPTION_FLAG_HIDDEN, G_OPTION_ARG_NONE, &opt_yes, "Does nothing, we always assume yes", NULL },
+  { "best", '\0', G_OPTION_FLAG_NONE, G_OPTION_ARG_NONE, &opt_best, "Try the best available package versions in transactions.", NULL },
```
Comment (jrohel) [inline dnf/dnf-main.c]: Not consistent. A dot at the end.

File: dnf/dnf-main.c [hunk 2]
```diff
@@ -283,7 +287,14 @@ main (int   argc,
 
       dnf_transaction_set_flags (txn, flags);
     }
-
+  if (!show_help && opt_best)
+    {
+        dnf_context_set_best(opt_best);
```
Comment (jrohel) [inline dnf/dnf-main.c]: Style. Remove two spaces.

File: dnf/dnf-main.c [hunk 3]
```diff
@@ -283,7 +287,14 @@ main (int   argc,
...
+    }
+  if (!show_help && opt_nobest && !opt_best)
+    {
+        dnf_context_set_best(FALSE);
```
Comment (jrohel) [inline dnf/dnf-main.c]: Style. Remove two spaces.

File: dnf/dnf-main.c [hunk 4]
```diff
@@ -283,7 +287,14 @@ main (int   argc,
 
       dnf_transaction_set_flags (txn, flags);
     }
-
+  if (!show_help && opt_best)
```
Comment (jrohel) [inline dnf/dnf-main.c]: I prefer this code below test and error message. And move it up into `if (!show_help)`:
```
      if (opt_best && opt_nobest)
        {
          error = g_error_new_literal(G_OPTION_ERROR,
                                      G_OPTION_ERROR_BAD_VALUE,
                                      "Argument --nobest is not allowed with argument --best");
          goto out;
        }
      if (opt_best)
        {
          dnf_context_set_best(TRUE);
        }
      else if (opt_nobest)
        {
          dnf_context_set_best(FALSE);
        }
```

**Comment by j-mracek** [comment]
@jrohel fixed. Thanks for review

**Comment by Conan-Kudo** [comment]
@rh-atomic-bot r+

**Pull request merged**